technical-reports/format/groups.md

```diff
@@ -34,251 +57,697 @@ A file MAY contain many tokens and they MAY be nested arbitrarily in groups like
+
+- Converting `"$extends": "{group}"` to `"$ref": "#/group"`
+- Applying JSON Schema 2020-12 `$ref` resolution with sibling property evaluation
+
+Tools implementing this specification MAY choose to:
+
+1. **Transform approach:** Convert `$extends` to `$ref` and use JSON Schema libraries
+2. **Direct implementation:** Implement the deep merge behavior described above
+3. **Hybrid approach:** Use JSON Schema for validation while maintaining design token syntax
+
+Regardless of implementation approach, the user-visible behavior MUST match the deep merge semantics described in this specification.
+
+#### `$deprecated`
+
+Groups MAY include an optional `$deprecated` property to mark the entire group as deprecated. This extends to all child tokens within the group unless explicitly overridden.
+
+| Value    | Explanation                                                       |
+| -------- | ----------------------------------------------------------------- |
+| `true`   | This group is deprecated (no explanation provided)                |
+| `string` | This group is deprecated AND this is an explanation               |
+| `false`  | This group is NOT deprecated (may override parent group defaults) |
+
+#### `$extensions`
+
+Groups MAY include an optional [`$extensions`](design-token#extensions) property where tools MAY add proprietary, user-, team- or vendor-specific data. Each tool MUST use a vendor-specific key whose value MAY be any valid JSON data.
+
+## Empty Groups
+
+Empty groups (groups containing no tokens or nested groups) are **explicitly permitted**. While they may appear to serve no immediate purpose, they:
+
+- Cause no harm to processing or validation
+- Support work-in-progress organization during token development
+- May contain metadata via group properties (`$description`, `$extensions`)
+- Provide placeholder structure for future token development
+
+<aside class="example">
+
+```json
+{
+  "experimental": {
+    "$description": "Tokens for experimental features - currently empty",
+    "$deprecated": "This group is being phased out"
+  }
+}
+```
+
+</aside>
+
+**Note on Token vs Group Ambiguity:**
+Objects without a `$value` property are interpreted as groups by definition. This can potentially create ambiguity in cases where a token lacks required properties (such as `$value` or a determinable type) and might be incorrectly parsed as an empty group. Tools SHOULD provide clear error messages when an object appears to be an incomplete token rather than an intentional empty group.
+
+## References and JSON Pointer Integration
+
+The current [token reference syntax](#references-and-json-pointer-integration) using curly braces (`{group.token}`) is maintained for backward compatibility and developer ergonomics. However, tools MAY also support JSON Pointer notation for advanced use cases.
+
+### Current Reference Syntax (Recommended)
+
+<aside class="example">
```

Comment:
Suggestion: I’ll stop commenting `title`s, but here and below, these could be example titles rather than section headings. IMO a section heading should have some meaningful text that stands alone apart from an example. Otherwise, the example should be nested under another heading.

This isn’t W3C styleguide or anything as far as I know. This is just personal preference (which, again, obligatory “I think we’re inconsistent about it”)

```suggestion
<aside class="example" title="Current Reference Syntax (Recommended)">
```